version: '3.7'

services:
  nginx-lua:
    image: fabiocicerchia/nginx-lua:${NGINX_LUA_TAG}
    restart: unless-stopped
    volumes:
      - ./configs/nginx.conf:/etc/nginx/nginx.conf:ro
      - ./configs/handler.lua:/etc/nginx/handler.lua:ro
      - ./episodes:/var/episodes:ro
      - ./static:/usr/share/nginx/html/:ro
    healthcheck:
      test: ['CMD', 'curl --fail http://localhost/ || exit 1']
      interval: 30s
      timeout: 3s
      retries: 3
    networks:
      - internal
  php:
    restart: unless-stopped
    image: php:${PHP_TAG}
    networks:
      - internal
    volumes:
      - ./episodes:/var/episodes:ro
      - ./static:/usr/share/nginx/html/:ro
  caddy:
    image: caddy:${CADDY_TAG}
    restart: unless-stopped
    ports:
      - "80:80"
      - "443:443"
      - "443:443/udp"
    networks:
      - internal
    volumes:
      - ./configs/Caddyfile:/etc/caddy/Caddyfile
      - caddy_data:/data
      - caddy_config:/config

volumes:
  caddy_data:
    external: true
  caddy_config:

networks:
  internal:
    driver: bridge